examples/move/scratch_off/sources/game.move

```diff
@@ -0,0 +1,549 @@
+            let target_prize_value = vector::pop_back(&mut value_of_prizes);
+            vector::push_back(&mut winning_tickets, PrizeStruct {
+                ticket_amount: target_prize_amount,
+                prize_value: target_prize_value
+            });
+
+            // prize_amount * prize_value to get the total we need.
+            // Note that this is a u64 * u64 so we could have overflow but for the purpose
+            // of this smart contract we do not need to consider this.
+            let target_amount = target_prize_amount * target_prize_value;
+            // Pull the required balance from the coin and stuff it into a balance.
+            balance::join(&mut prize_pool, coin::into_balance(coin::split(&mut coin, target_amount, ctx)));
+            winning_ticket_count = winning_ticket_count + target_prize_amount;
+            idx = idx + 1;
+        };
+
+        let new_store = ConvenienceStore<Asset> {
+            id: object::new(ctx),
+            creator: tx_context::sender(ctx),
+            prize_pool,
+            winning_tickets, 
+            original_ticket_count: winning_ticket_count,
+            tickets_issued: 0, 
+            tickets_evaluated: 0,
+            player_metadata: table::new(ctx),
+            leaderboard: LeaderBoard {
+                lowest_sui_won: 0,
+                max_players: max_players_in_leaderboard,
+                leaderboard_players: vector[],
+                leaderboard_player_metadata: table::new(ctx),
+            },
+            public_key,
+        };
+        transfer::public_transfer(StoreCap {
+            id: object::new(ctx),
+            store_id: object::id(&new_store)
+        }, tx_context::sender(ctx));
+
+        transfer::share_object(new_store);
+        coin
+    }
+
+    /// Initializes a ticket and sends it to someone.
+    public fun send_ticket<Asset>(
+        store_cap: &StoreCap,
+        player: address,
+        store: &mut ConvenienceStore<Asset>,
+        ctx: &mut TxContext
+    ) {
+        assert!(store_cap.store_id == object::uid_to_inner(&store.id), ENotAuthorizedEmployee);
+        assert!(store.tickets_issued < store.original_ticket_count, ENoTicketsLeft);
+        store.tickets_issued = store.tickets_issued + 1;
+        let ticket = Ticket {
+            id: object::new(ctx),
+            convenience_store_id: object::uid_to_inner(&store.id),
+            player
+        };
+        transfer::public_transfer(ticket, player);
+    }
+
+    /// Calls evaluate ticket by adding it as a dynamic field to the store
+    /// We actually have a choice here to make users send tickets to our address and write
+    /// the backend to listen to those, or we can list these objects as a dof in the shared_object
+    public fun evaluate_ticket<Asset>(
+        ticket: Ticket,
+        store: &mut ConvenienceStore<Asset>,
+        ctx: &mut TxContext
+    ): ID {
+        assert!(store.tickets_issued <= store.original_ticket_count, ENoTicketsLeft);
```

Comment:
I suppose that is a sanity check? considering we can't mint a `Ticket` above the limit in the `send_ticket` function.

Reply:
Yes purely sanity checks which I think should just exist in the code